webpack/index.d.ts

```diff
@@ -506,15 +520,63 @@ declare namespace webpack {
     }
 
     interface DefinePluginStatic {
-        new (definitions: any): Plugin;
+        new (definitions: {[key: string]: any}): Plugin;
     }
 
     interface ProvidePluginStatic {
-        new (definitions: any): Plugin;
+        new (definitions: {[key: string]: any}): Plugin;
     }
 
     interface SourceMapDevToolPluginStatic {
-        new (options: any): Plugin;
+        new (options?: SourceMapDevToolPluginOptions): Plugin;
+    }
+
+    // if string | false | null, maps to the filename option
+    type SourceMapDevToolPluginOptions = string | false | null | {
+        // output filename pattern (false/null to append)
+        filename?: string | false | null,
```

Comment:
This is more like "string | falsy", but I don't think people should be giving `0` as the filename.